### PR TITLE
Add popupFormat option to GeoSearchControl

### PR DESCRIPTION
**Pull request by ville6000**
Allow users to customize popup text

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -249,18 +249,19 @@ There are a number of options to adjust the way results are visualized.
...
+  marker: {                             // optional: L.Marker    - default L.Icon.Default
     icon: new L.Icon.Default(),
     draggable: false,
   },
-  maxMarkers: 1,                  // optional: number      - default 1
-  retainZoomLevel: false,         // optional: true|false  - default false
-  animateZoom: true,              // optional: true|false  - default true
-  autoClose: false,               // optional: true|false  - default false
-  searchLabel: 'Enter address'    // optional: string      - default 'Enter address'
+  popupFormat: result => result.label,  // optional: function    - default returns result label
```
Comment (smeijer) [inline readme.md]: Please change the signature to: ` ({ query, result })`.

File: src/leafletControl.js [hunk 2]
```diff
@@ -10,6 +10,7 @@ const defaultOptions = () => ({
   style: 'button',
   showMarker: true,
   showPopup: false,
+  popupFormat: result => result.label,
```
Comment (smeijer) [inline src/leafletControl.js]: Please change signature to `({ query, result }) => result.label`

File: src/leafletControl.js [hunk 3]
```diff
@@ -220,9 +221,15 @@ const Control = {
...
 
   addMarker(result) {
-    const { marker: options, showPopup } = this.options;
+    const { marker: options, showPopup, popupFormat } = this.options;
     const marker = new L.Marker([result.y, result.x], options);
-    marker.bindPopup(result.label);
+    let popupLabel = result.label;
+
+    if (typeof popupFormat === 'function') {
+      popupLabel = popupFormat(result);
```
Comment (smeijer) [inline src/leafletControl.js]: Change `popupFormat` signature to `popupFormat({ query, result })`. This does require the `addMarker` method to be called with  addMarker(result, query)`, as well as a similar change to `showResult`, but gives the end user ultimate flexibility in styling his popup.

**Comment by smeijer** [comment]
Thanks, seems to be related to #89 . Could you address the suggested changes, so we can merge it?

**Comment by ville6000** [comment]
@smeijer I added the query parameter

**Pull request merged**